### PR TITLE
Moderation features testing

### PR DESCRIPTION
**Pull request by sbinlondon**
Testing new moderation features

* tests Collective settings validation relating to `settings.moderation.rejectedCategories`
* `canReject` function: no tests because for now it's the exact same permissions as `canRefund`
* tests `rejectTransaction` v2 mutation

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
Looking good!

> canReject function: no tests because for now it's the exact same permissions as canRefund

The main purpose of tests is to detect breaking changes, so in such situation it's fine (and better) to copy paste and adapt the tests from `canRefund` - not to make sure that the feature works today, but to make sure it won't break in the future (ie. if we refactor the code for `canReject`).

Moreover, I see that `canRefund` and `canReject` are not using a common helper (they're duplicated code), so it would be easy to break one and not the other.

Finally, no update have been made on `canRefund` for testing the "30 days" limit. We should add a `Collective admins can only refund during 30 days` test on `canRefund` + `canReject`

**Comment by sbinlondon** [comment]
> Finally, no update have been made on canRefund for testing the "30 days" limit. We should add a Collective admins can only refund during 30 days test on canRefund + canReject

I thought I had added that, my apologies. Will do now!

**Review by znarf — APPROVED** [review]

File: test/server/graphql/common/transactions.test.js [hunk 1]
```diff
@@ -78,9 +89,35 @@ describe('server/graphql/common/transactions', () => {
...
       expect(await canRefund(transaction, undefined, fromCollectiveAccountantReq)).to.be.false;
+      expect(await canRefund(transaction, undefined, hostAdminReq)).to.be.true;
+      expect(await canRefund(transaction, undefined, rootAdminReq)).to.be.true;
+      expect(await canRefund(oldTransaction, undefined, hostAdminReq)).to.be.true;
+      expect(await canRefund(oldTransaction, undefined, rootAdminReq)).to.be.true;
+    });
+
+    it('can refund as admin of the receiving collective only if transaction < 30 days old', async () => {
       expect(await canRefund(transaction, undefined, collectiveAdminReq)).to.be.true;
+      expect(await canRefund(oldTransaction, undefined, collectiveAdminReq)).to.be.false;
```
Comment (Betree) [inline test/server/graphql/common/transactions.test.js:100]: (Not a requested change) Would have been great to test that host admins are not impacted by this

**Review by Betree — COMMENTED** [review]
Looks good :+1:

**Pull request merged**